src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+    }
+
+    public MyLinkedList(T element) {
+        Node<T> firstNode = new Node<>(null, element, null);
+        head = tail = firstNode;
+        size = 1;
+    }
+
+    private void checkBounds(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of LinkedList bonds");
+        }
+    }
+
+    private void insertNode(Node<T> targetNode, T value) {
+        Node<T> nodeToInsert = new Node<>(targetNode.previous, value, targetNode);
+        if (targetNode.previous == null) {
+            targetNode.previous = nodeToInsert;
+            head = nodeToInsert;
+        } else {
+            targetNode.previous.next = nodeToInsert;
+            targetNode.previous = nodeToInsert;
+        }
+        size++;
+    }
+
+    private void unlink(Node<T> nodeToRemove) {
+        if (size == 1) {
+            head = tail = null;
+        } else if (nodeToRemove == head) {
+            nodeToRemove.next.previous = null;
+            head = nodeToRemove.next;
+        } else if (nodeToRemove == tail) {
+            nodeToRemove.previous.next = null;
+            tail = nodeToRemove.previous;
+        } else {
+            nodeToRemove.previous.next = nodeToRemove.next;
+            nodeToRemove.next.previous = nodeToRemove.previous;
+        }
+        nodeToRemove.next = nodeToRemove.previous = null;
+        nodeToRemove.storedItem = null;
+        --size;
+    }
+
+    private Node<T> findNode(int index) {
+        Node<T> headHandler = head;
+        Node<T> tailHandler = tail;
```

Comment:
you don't need two variables here 
```suggestion
        Node<T> currentNode = head;
```

Reply:
done. вышло столько же строчек, так как return вынес за циклы)